core/embed/trezorhal/stm32f4/dma2d_bitblt.c

```diff
@@ -113,20 +116,31 @@ static void dma2d_config_clut(uint32_t layer, gfx_color_t fg, gfx_color_t bg) {
   volatile uint32_t* clut =
       layer ? dma2d_handle.Instance->FGCLUT : dma2d_handle.Instance->BGCLUT;
 
-  if (fg != cache[layer].c_fg || bg != cache[layer].c_bg) {
+  if (fg != cache[layer].c_fg || bg != cache[layer].c_bg ||
+      alpha != cache[layer].alpha || blend != cache[layer].blend) {
     cache[layer].c_fg = fg;
     cache[layer].c_bg = bg;
-
-    for (int step = 0; step < GRADIENT_STEPS; step++) {
-      clut[step] = gfx_color32_blend_a4(fg, bg, step);
+    cache[layer].alpha = alpha;
+    cache[layer].blend = blend;
+
+    if (blend) {
+      gfx_color32_t fg32 = gfx_color16_to_color32(fg);
```

Comment:
No sure if it might work, but::
- do not incorporate 8-bit alpha into the gradient
- set `AlphaMode` in layer configuration to `DMA2D_COMBINE_ALPHA`
- set `InputAlpha` in layer configuration to 8-bit alpha to `bb->src_alpha`

Reply:
it works like this: https://github.com/trezor/trezor-firmware/commit/2f741408ba9e087a6dc87a94c53c81831e7837d5
so setting alpha in the CLUT with alpha only from gradient, the InputAlpha is set to bb->src_alpha and combination is used. The gradient function still needs blend parameter because for blending, color is constant and only alpha changes, but for copying the alpha needs to be full and color is adjusted.

Reply:
yeah its better like that, i cherry picked your commit https://github.com/trezor/trezor-firmware/commit/fb5f691d2132200128645924070d179a9f76f690

one thing that i changed was this https://github.com/trezor/trezor-firmware/commit/0a6558c1f8cbd7daf043355e90a7fd8484433e7c - the background color should be irrelevant when blending